### PR TITLE
fix(Scripts/NPC): Arcanite Dragonling

### PR DESCRIPTION
**Pull request by grimgravy**
## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12284

## SOURCE:

https://www.youtube.com/watch?v=xi2Rvi4VfCs
## Tests Performed:

- Tested in game

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
It works. It suffers from the common problems nearly all these summons have but it works 😅 👍

File: src/server/scripts/World/npcs_special.cpp [hunk 1]
```diff
@@ -2542,6 +2542,63 @@ class npc_venomhide_hatchling : public CreatureScript
...
+
+    void Reset() override
+    {
+        events.Reset();
+    }
+
+    void EnterCombat(Unit* /*who*/) override
+    {
+        events.ScheduleEvent(EVENT_FLAME_BUFFET, 4000);
+        events.ScheduleEvent(EVENT_FLAME_BREATH, 12000);
```
Comment (Nefertumm) [inline src/server/scripts/World/npcs_special.cpp:2571]:
```suggestion
        events.ScheduleEvent(EVENT_FLAME_BUFFET, 4s);
        events.ScheduleEvent(EVENT_FLAME_BREATH, 12s);
```

File: src/server/scripts/World/npcs_special.cpp [hunk 2]
```diff
@@ -2542,6 +2542,63 @@ class npc_venomhide_hatchling : public CreatureScript
...
+
+    void UpdateAI(uint32 diff) override
+    {
+        events.Update(diff);
+
+        switch (events.ExecuteEvent())
+        {
+            case EVENT_FLAME_BUFFET:
+                me->CastSpell(me->GetVictim(), SPELL_FLAME_BUFFET, false);
+                events.RepeatEvent(12000);
```
Comment (Nefertumm) [inline src/server/scripts/World/npcs_special.cpp:2587]:
```suggestion
                DoCastVictim(SPELL_FLAME_BUFFET);
                events.Repeat(12s);
```

File: src/server/scripts/World/npcs_special.cpp [hunk 3]
```diff
@@ -2542,6 +2542,63 @@ class npc_venomhide_hatchling : public CreatureScript
...
+                break;
+            case EVENT_FLAME_BREATH:
+                me->CastSpell(me->GetVictim(), SPELL_FLAME_BREATH, false);
+                events.RepeatEvent(24000);
```
Comment (Nefertumm) [inline src/server/scripts/World/npcs_special.cpp:2591]:
```suggestion
                DoCastVictim(SPELL_FLAME_BREATH);
                events.Repeat(24s);
```

File: src/server/scripts/World/npcs_special.cpp [hunk 4]
```diff
@@ -2542,6 +2542,63 @@ class npc_venomhide_hatchling : public CreatureScript
...
+                break;
+        }
+
+        if (!UpdateVictim())
+            return;
```
Comment (Nefertumm) [inline src/server/scripts/World/npcs_special.cpp:2596]: Is this okay to be down here?

**Review by Nefertumm — CHANGES_REQUESTED** [review]

File: src/server/scripts/World/npcs_special.cpp (hunk 4 above)
Comment (grimgravy) [reply src/server/scripts/World/npcs_special.cpp:2596]: it doesn't change anything

**Review by Nefertumm — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**